mux.go

```diff
@@ -197,8 +193,14 @@ func (r *Router) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	var handler http.Handler
 	if r.Match(req, &match) {
 		handler = match.Handler
-		req = requestWithVars(req, match.Vars)
-		req = requestWithRoute(req, match.Route)
+		if handler == nil {
+			// The default handlers do not make use of the context values.
+		} else if r.omitRouteFromContext {
+			// Only populate the match vars (if any) into the context.
+			req = requestWithVars(req, match.Vars)
+		} else {
+			req = requestWithRouteAndVars(req, match.Route, match.Vars)
+		}
```

Comment:
Can we refactor this block as follows? I understand the reasoning here, but having empty branches can be a bit confusing and mask the code's real intentions.
```suggestion
		if handler != nil {
			if r.omitRouteFromContext {
				// Only populate the match vars (if any) into the context.
				req = requestWithVars(req, match.Vars)
			} else {
				req = requestWithRouteAndVars(req, match.Route, match.Vars)
			}
		}
```

Reply:
Sure, done via 83efd146aee45de91454b098fcdeacf51d04658b (includes an extra comment)